### ISSUE TITLE
DeaDBeeF 1.9.5 crash; can't find x64lib libstdc++.so.6 from box64 in lib directory

### ISSUE THREAD

**Issue by llirium** [ptitSeb/box64#707]
Not sure what I'm doing wrong.  The `libstdc++.so.6` library is in the proper `lib` directory for the DeaDBeeF, it's executable too.

```
$ box64 deadbeef
Dynarec for ARM64, with extension: ASIMD CRC32 PageSize:4096
Box64 with Dynarec v0.1.8 nogit built on Jan  1 1980 00:00:00
Using default BOX64_LD_LIBRARY_PATH: ./:lib/:lib64/:x86_64/:bin64/:libs64/
Using default BOX64_PATH: ./:bin/
Counted 71 Env var
Looking for deadbeef
Using native(wrapped) libdl.so.2
Using native(wrapped) libm.so.6
Using native(wrapped) libpthread.so.0
Using emulated /home/llirium/Builds/deadbeef-1.9.5/lib/libdispatch.so
Using emulated /home/llirium/Builds/deadbeef-1.9.5/lib/libBlocksRuntime.so
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) librt.so.1
Error loading needed lib libstdc++.so.6
Error loading one of needed lib
Error initializing needed lib libdispatch.so
Error loading one of needed lib
Error: loading needed libs in elf /home/llirium/Builds/deadbeef-1.9.5/deadbeef
```

**Comment by ptitSeb** [comment]
I wonder how you install box64, because that libs should come package with it.
Anyway, use `LD_LIBRARY_PATH` or `BOX64_LD_LIBRARY_PATH` to add folder to box64 library search path.

**Comment by llirium** [comment]
Not experienced enough to mess with `LD_LIBRARY_PATH`, so I tried your second suggestion, still no luck.  I'm running NixOS, it's set up quite differently in comparison to most distros, if that becomes relevant.

The attached file is the output of this command:
`BOX64_LOG=2 BOX64_LD_LIBRARY_PATH="/home/llirium/Builds/deadbeef-1.9.5/lib" box64 deadbeef > box64_deadbeef_loglevel2_2023-04-18.txt`

[box64_deadbeef_loglevel2_2023-04-18.txt](https://github.com/ptitSeb/box64/files/11264597/box64_deadbeef_loglevel2_2023-04-18.txt)

**Comment by ptitSeb** [comment]
Did you put the x86_64 version of `libstdc++.so.6` in `/home/llirium/Builds/deadbeef-1.9.5/lib`?

**Comment by llirium** [comment]
Yes, `libstdc++.so.6` is there:

```
[llirium@nixospi4:~/Builds/deadbeef-1.9.5]$ tree lib/
[drwxr-xr-x llirium    users    1.7M]  lib//
├── [-rw-r--r-- llirium    users     14K]  libBlocksRuntime.so
├── [lrwxrwxrwx llirium    users      16]  libcurl.so -> libcurl.so.4.7.0*
├── [lrwxrwxrwx llirium    users      16]  libcurl.so.4 -> libcurl.so.4.7.0*
├── [-rwxr-xr-x llirium    users    519K]  libcurl.so.4.7.0*
├── [-rw-r--r-- llirium    users    403K]  libdispatch.so
├── [lrwxrwxrwx llirium    users      18]  libmbedcrypto.so -> libmbedcrypto.so.2*
├── [lrwxrwxrwx llirium    users      23]  libmbedcrypto.so.2 -> libmbedcrypto.so.2.7.19*
├── [-rwxr-xr-x llirium    users    392K]  libmbedcrypto.so.2.7.19*
├── [lrwxrwxrwx llirium    users      16]  libmbedtls.so -> libmbedtls.so.10*
├── [-rwxr-xr-x llirium    users    186K]  libmbedtls.so.2.7.19*
├── [lrwxrwxrwx llirium    users      20]  libmbedtls.so.10 -> libmbedtls.so.2.7.19*
├── [lrwxrwxrwx llirium    users      16]  libmbedx509.so -> libmbedx509.so.0*
├── [lrwxrwxrwx llirium    users      21]  libmbedx509.so.0 -> libmbedx509.so.2.7.19*
├── [-rwxr-xr-x llirium    users     95K]  libmbedx509.so.2.7.19*
└── [-rwxr-xr-x llirium    users    138K]  libstdc++.so.6*
```

Downloaded the library itself from here a day ago: [https://github.com/ptitSeb/box64/blob/main/x64lib/libstdc++.so.6](https://github.com/ptitSeb/box64/blob/main/x64lib/libstdc++.so.6)

**Comment by ptitSeb** [comment]
You also need `libgcc_s.so.1` probably.

**Comment by llirium** [comment]
Apologies, I think the problem is on my side and I didn't realize it because it may have been an improper understanding on how NixOS works:  https://blog.thalheim.io/2022/12/31/nix-ld-a-clean-solution-for-issues-with-pre-compiled-executables-on-nixos/

tl;dr:  NixOS does not currently play nicely with pre-compiled executables that are not properly integrated with the OS itself (up until NixOS version 23.05 becomes the stable release, hopefully).

Thanks for the help all the same, sorry for not realizing sooner.
